src/adapter.ts

```diff
@@ -230,6 +234,19 @@ export default class JestTestAdapter implements TestAdapter {
     }
   }
 
+  private getOrCreateOutputWindowForProject(project: ProjectRootNode): vscode.OutputChannel {
+    const channelName = `Jest (${project.config.projectName})`;
+    const channel = this.outputChannels.find(c => c.name === channelName);
```

Comment:
I understand that this was why you changed `disposables ` to be `outputChannels`, and it makes sense.  I still feel like keeping `disposables` was a cleaner solution for managing those - if anyone adds any more, they now need to remember to add them to the `dispose` method.

On a related note, I believe VSCode provides you with its own `disposables` collection during `activate` that you can append to and VSCode itself will clean them up: [`context.subscriptions`](https://code.visualstudio.com/api/get-started/extension-anatomy#extension-entry-file).  Perhaps you can start pushing disposables there.

Reply:
You are right about the context.subscriptions and we are already using that in the main.ts method to tear down the whole extension.  I personally like the tree approach where each object is responsible for calling dispose on any child objects that it is responsible for.  Then you just have to call dispose on the top of the tree and everything works correctly.

I did go back and forth on whether to replace the disposables collection or not.  I might revert it back to the way it was.  Either way any future dev will have to remember to do something to ensure that their objects are disposed.

Reply:
I completely missed the fact that this object is not the top-level extension object.  It makes perfect sense for this object to track its own disposables, and then this object itself gets disposed of by something above it.